codewind-che-sidecar/scripts/kube/codewind_template.yaml

```diff
@@ -155,3 +159,117 @@ spec:
           limits:
             memory: 2Gi
             cpu: 500m
+
+---
+
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: codewind-ingress-WORKSPACE_ID_PLACEHOLDER
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+  ownerReferences:
+    - apiVersion: apps/v1
+      blockOwnerDeletion: true
+      controller: true
+      kind: ReplicaSet
+      name: OWNER_REFERENCE_NAME_PLACEHOLDER
+      uid: OWNER_REFERENCE_UID_PLACEHOLDER
+spec:
+  rules:
+  - host: codewind-WORKSPACE_ID_PLACEHOLDER.CHE_INGRESS_DOMAIN_PLACEHOLDER
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: codewind-WORKSPACE_ID_PLACEHOLDER
```

Comment:
Does Codewind need to be exposed over ingress? Ingress is generally useful for services that users would want to access externally, but users won't be accessing Codewind directly, but rather via Theia/Che. For communication between Codewind and the performance dashboard, you can use `<serviceName>:<internal-port>`

Will the performance dashboard need an ingress instead?

Reply:
The performance page loads from Theia in a new browser window and needs a way "back in" to portal to talk to its apis.   The performance page itself is also loaded via the PFE ingress by way of a reverse proxy into the performance container.   That avoids any CORS issues since both UI and APIs talk to the same host. The performance container itself does not need an ingress.

Reply:
Ah, that sounds good to me then.